feat: upgrade go version to 1.22

FrankYang0529 opened the pull request:
https://github.com/harvester/harvester/issues/6160

FrankYang0529 committed:
feat: upgrade go version to 1.22

Signed-off-by: PoAn Yang <poan.yang@suse.com>

FrankYang0529 committed:
feat(Dockerfie): bump golangci-lint to v1.58.2

Signed-off-by: PoAn Yang <poan.yang@suse.com>

Vicente-Cheng reviewed (approved):
LGTM, just a nit.

Could we add more info about why we remove the validate-ci?
(On the corresponding commit)

FrankYang0529 committed:
feat: remove validate-ci

Currently, rancherd doesn't use any `go generate`, so we don't need
validate-ci. Also, go 1.22 shows `no Go files ...` when running `go
generate`.

Signed-off-by: PoAn Yang <poan.yang@suse.com>

FrankYang0529 commented:
> Could we add more info about why we remove the validate-ci?
> (On the corresponding commit)

Yes, updated it. Thanks.

ibrokethecloud reviewed (approved):
lgtm. thanks.
